src/patterns/ec2-app/base.ts

```diff
@@ -470,6 +511,45 @@ export class GuEc2App extends Construct {
       open: access.scope === AccessScope.PUBLIC,
     });
 
+    if (props.googleAuth?.enabled) {
+      const configPrefix = `${scope.stage}/${scope.stack}/${app}`;
+      const clientIdPath = props.googleAuth.clientIdPath ?? `/${configPrefix}/googleClientID`;
+
+      const clientId = StringParameter.fromStringParameterAttributes(this, "clientID", {
+        parameterName: clientIdPath,
+      }).stringValue;
+
+      // Unfortunately, Cloudformation doesn't support directly using secret
+      // Parameter Store values. But it is possible to use Secrets Manager.
+      const secretPath = props.googleAuth.clientSecretPath ?? `${configPrefix}/clientSecret`;
+      const clientSecret = SecretValue.secretsManager(secretPath);
+
+      const authAction = ListenerAction.authenticateOidc({
+        next: ListenerAction.forward([targetGroup]),
+        clientId: clientId,
+        clientSecret: clientSecret,
+        scope: "openid email",
+
+        // See the `hd` section of
+        // https://developers.google.com/identity/protocols/oauth2/openid-connect#authenticationuriparameters.
+        // Note, this is NOT sufficient to ensure access is limited to Guardian
+        // emails. Users should also ensure set 'User Type' to 'Internal' in the
+        // Google Console Consent Screen to ensure only @guardian.co.uk members
+        // can access your app and ensure (through Security Groups) that only
+        // the ALB is able to access the EC2 instances.
+        authenticationRequestExtraParams: { hd: "guardian.co.uk" },
+
+        authorizationEndpoint: "https://accounts.google.com/o/oauth2/v2/auth",
+        issuer: "https://accounts.google.com",
+        tokenEndpoint: "https://oauth2.googleapis.com/token",
+        userInfoEndpoint: "https://openidconnect.googleapis.com/v1/userinfo",
+      });
+
+      listener.addAction("auth", { action: authAction });
+
+      Tags.of(loadBalancer).add(MetadataKeys.CDK_FEATURE, "google-auth");
```

Comment:
What's the purpose of this tag?

Reply:
To support possible tracking in future. It's not a construct so our generic metadating tracking doesn't work unfortunately, but it is still something that would be good to track in the future.

Reply:
> To support possible tracking in future.

IIUC we don't have mechanisms to make use of this tag right now? That is, should we decouple this change from the OIDC part, whilst tracking is undefined?